app/src/main/java/io/gnosis/kouban/ui/transaction/details/TransactionDetailsViewModel.kt

```diff
@@ -45,30 +43,51 @@ class TransactionDetailsViewModel(
         }
 
     private fun Pair<Solidity.Address?, ServiceSafeTx>.toDetails(): List<Any> {
-        return listOf(
-            first!!,
-            second.tx.buildTransactionTypeView(first),
-            second.tx.to,
-            LabelDescription(
-                R.string.transaction_details_network_fees_label,
-                SpannableString(second.execInfo.fees.shiftedString(ETH_TOKEN_INFO.decimals, decimalsToDisplay = ETH_TOKEN_INFO.decimals))
-            ),
-            LabelDate(R.string.transaction_details_timestamp_label, dateInSecs = transaction.timestamp),
-            Link(transaction.executionHash!!, R.string.view_transaction_on),
-            LabelDescription(R.string.transaction_details_raw_data_label, SpannableString(second.tx.data))
-        )
+        val decodedData = decode(second.tx)
+        return mutableListOf<Any>().apply {
+            add(first!!)
```

Comment:
if first should not be nullable reflect it in the signature Pair<Solidity.Address, ServiceSafeTx>.toDetails()

Reply:
I am forced by the return type from which I fetch `first`. That is `addressManager.getSafeAddress()`

Reply:
I agree with vitaly, if we expect it to never be null make it non null and handle it gracefully when calling this method

Reply:
Either way, this was running from within a `runCatching` block, so the app wouldn't crash. I changed the exception type with a better error message for the logs.